### DISCUSSION TITLE
Reorder singles in the CP

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9177] [Ideas]
> Created by: Andy Harris (andy@pepperdigital.com) on 2015/03/19 11:03:06 +0000
> Votes at time of UserVoice import: 6

-----------------------------------

We have some sites with quite a lot of singles, and ordering by date or title doesn't really have any relevance - it'd be nice to be able to control the order, so home is always at the top and the rest matched the site itself as closely as possible.

Purely a cosmetic CP thing.

**Comment by angrybrad** [comment]
> Posted by Adam Whitehouse (awhitehouse104@gmail.com) on 2016/12/05 21:21:02 +0000
-----------------------------------

I&#39;m not a fan of the singles all being grouped in a single tab together anyway. The fact that they&#39;re all singles is irrelevant to the content editor. That being said, this suggestion would certainly be an improvement. As you say, there&#39;s not much meaning in ordering by title or date.

Daryl, I&#39;m not sure what point you&#39;re trying to make?

**Comment by angrybrad** [comment]
> Posted by Daryl (craft@codeknight.co.uk) on 2015/03/12 09:36:29 +0000
-----------------------------------

What sort of singles are you using? Are you sure they shouldn&#39;t be in a Structure instead?

**Comment by leowebguy** [comment]
agree, let's have an option to organize singles in blocks

**Comment by ryanmasuga** [comment]
Every time we have a site with any number of singles, we see them in a natural hierarchy - and it's different for every site. Home at top, maybe "About" is next, and so on. Every time I want to drag/drop to reorder them like we do for Structures, just to be organized. I think this would be a nice enhancement, if Singles are going to remain their own separate list.

**Comment by thupsi** [comment]
This probably feels like a stop-gap solution till a kind of "Unified Content View" arrives, but here goes anyway 🙂 :

As P&T are aware, the current handling of "singles" in the CP poses some problems, but I'm going to focus on the use case of a very simple brochure site. 

1. Each page on the front-end demands a unique field layout, so I created a single for every page.
2. Each item on the global navigation corresponds to a page.
3. I want a specific order for the pages, and I ideally want that to be sortable by the client.

This is a very simple setup and yet the page sorting cannot be solved in a simple and elegant way:

1. I can create a **"navigation" structure**, but in this case that seems like unnecessary duplication. The client would have a "singles" item in the sidebar (which I have renamed to "pages") and a "navigation" item, with exactly the same names in the list, just in a different order.
2.  I could maybe improve on the above by using a **global set**. Cleaner UI. Still duplicating things though, and a step back for discoverability.
3. I could create a "Pages" structure and **use entry types instead of singles** (a popular approach to solve such issues), but honestly this also feels like going against the grain, and you have to implement some kind of hack to prevent switching of entry types etc. Not to mention that now the "Singles" are treated like normal entries and are not propagated via Project Config.
4. I can use the "**Expanded Singles**" plugin, so then I can sort the singles in the sidebar, but this poses its own set of problems: You either have to click on each sidebar item and THEN click on the actual entry, OR (with the "redirect to entry" option), you lose the ability to open the entries in slideouts.
5. Finaly, I can add a "sorting" numerical field and use that. This feels like the best compromise, but still not ideal: It doesn't enable drag and drop, and adds a little clutter to the entry form.

My thinking is that the only missing piece for this particular case is the ability to **treat the "Singles" group like a structure in the UI and enable manual sorting by drag and drop**.

I'm aware that this does not solve more complex cases, but it would be _awesome_ to have for basic scenarios.

It's also entirely possible that I'm missing a better solution, so I'm _very open_ to suggestions on how to approach this with current functionality.
